File: Data Structures & Algorithms/binary-tree-diameter/submission-7.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
var ans int
func dfs(root *TreeNode) int{
	if root == nil{
		return 0
	}
	left := 1 + dfs(root.Left)
	right := 1 + dfs(root.Right)
	ans = max(left+right - 2, ans) // remove duplicate node, and remove 1 more for edge count
	return max(left, right)
}
func diameterOfBinaryTree(root *TreeNode) int {
	ans = 0
	dfs(root)
	return ans
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-7.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        self.ans = []

        def rec(root,depth):
            if root is None:
                return
            
            if len(self.ans) == depth:
                self.ans.append(root.val)
            else:
                self.ans[depth] = root.val
            rec(root.left,depth+1)
            rec(root.right,depth+1)
        rec(root,0)
        return self.ans
            
        